### PR TITLE
SONAR-18977 Deprecates embedded PostgreSQL Chart

### PR DESCRIPTION
**Pull request by davividal**

### CODE REVIEW

File: charts/sonarqube-dce/values.schema.json [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {
     "$schema": "https://json-schema.org/draft/2020-12/schema",
     "properties": {
+        "postgresql": {
+            "type": "object",
+            "deprecated": true,
+            "$comment": "(DEPRECATED) Please use an external database instead of the embedded one. Please see https://artifacthub.io/packages/helm/sonarqube/sonarqube#jdbc-overwrite for more information"
```
Comment (carminevassallo) [inline charts/sonarqube-dce/values.schema.json:7]: There is a wrong link here.

File: charts/sonarqube-dce/README.md [hunk 2]
```diff
@@ -415,7 +415,9 @@ The following table lists the configurable parameters of the SonarQube chart and
 | `jdbcOverwrite.jdbcSecretName`        | Alternatively, use a pre-existing k8s secret containing the DB password                                                                                      | `None`                                                        |
 | `jdbcOverwrite.jdbcSecretPasswordKey` | If the pre-existing k8s secret is used this allows the user to overwrite the 'key' of the password property in the secret                                    | `None`                                                        |
 
-### Bundled Postgres Chart
+### Bundled Postgres Chart (DEPRECATED)
```
Comment (carminevassallo) [inline charts/sonarqube-dce/README.md:418]: Can we take the chance to fix the typo here (and elsewhere).  `Postgres` -> ` PostgreSQL`?

File: charts/sonarqube-dce/templates/NOTES.txt [hunk 3]
```diff
@@ -17,3 +17,7 @@
   echo "Visit http://127.0.0.1:8080 to use your application"
   kubectl port-forward $POD_NAME 8080:{{ .Values.service.externalPort }} -n {{ .Release.Namespace }}
 {{- end }}
+{{ if .Values.postgresql.enabled }}
+WARNING: The embedded PostgreSQL is intended for evaluation only, it is DEPRECATED, and it will be REMOVED in a future release.
+         Please see https://artifacthub.io/packages/helm/sonarqube/sonarqube#production-use-case for more information.
```
Comment (carminevassallo) [inline charts/sonarqube-dce/templates/NOTES.txt:22]: Please see -> Please visit

File: charts/sonarqube/templates/NOTES.txt [hunk 4]
```diff
@@ -20,3 +20,7 @@
   echo "Visit http://127.0.0.1:8080 to use your application"
   kubectl port-forward $POD_NAME 8080:{{ .Values.service.externalPort }} -n {{ .Release.Namespace }}
 {{- end }}
+{{ if .Values.postgresql.enabled }}
+WARNING: The embedded PostgreSQL is intended for evaluation only, it is DEPRECATED, and it will be REMOVED in a future release.
+         Please see https://artifacthub.io/packages/helm/sonarqube/sonarqube#production-use-case for more information.
```
Comment (carminevassallo) [inline charts/sonarqube/templates/NOTES.txt:25]: see above.

File: charts/sonarqube/values.schema.json [hunk 5]
```diff
@@ -4,6 +4,11 @@
         "replicaCount"
     ],
     "properties": {
+        "postgresql": {
+            "type": "object",
+            "deprecated": true,
+            "$comment": "(DEPRECATED) Please use an external database instead of the embedded one. Please see https://artifacthub.io/packages/helm/sonarqube/sonarqube#jdbc-overwrite for more information"
```
Comment (carminevassallo) [inline charts/sonarqube/values.schema.json:10]: see above

File: charts/sonarqube-dce/README.md (hunk 2 above)
Comment (davividal) [reply charts/sonarqube-dce/README.md]: That is not really a typo: https://www.postgresql.org/docs/current/history.html#HISTORY-POSTGRESQL. :)

But I also prefer PostgreSQL. :)

**Review by carminevassallo — APPROVED** [review]

**Comment by carminevassallo** [comment]
LGTM! Just a minor (non-blocking) comment on the description of the PostgreSQL parameters. We now use sometimes `Postgresql` and `PostgreSQL`, we could be consistent instead.

**Pull request merged**